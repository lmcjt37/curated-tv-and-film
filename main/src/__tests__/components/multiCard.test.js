import React from 'react';
import ReactDOM from 'react-dom';

import MultiCard from '../../components/multiCard';

import { createRender } from '@material-ui/core/test-utils';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const itemMockZero = {
  type: 'tv_show',
  title: 'The Walking Dead',
  genre: ['Drama', 'Horror', 'Action'],
  content: []
};

const itemMockOne = {
  type: 'tv_show',
  title: 'The Walking Dead',
  genre: ['Drama', 'Horror', 'Action'],
  content: [
    {
      description:
        'Following on from where Rick Grimes wakes up from a coma and tries to find his family. He gets to Atlanta, surrounded by walkers, he is thrown from his horse and ends up in a tank. This is where someone on the walkie talkie, Glenn, helps him out of the tight situation.',
      season: 1,
      episode: 2,
      episode_title: 'Guts',
      imdb: 'https://www.imdb.com/title/tt1628064/?ref_=ttep_ep2',
      url: 'https://www.youtube.com/watch?v=ZxmWhlidAZ8',
      thumbnail: './assets/images/walking-dead.jpg'
    }
  ]
};

const itemMockTwo = {
  type: 'tv_show',
  title: 'The Walking Dead',
  genre: ['Drama', 'Horror', 'Action'],
  content: [
    {
      description:
        'Following on from where Rick Grimes wakes up from a coma and tries to find his family. He gets to Atlanta, surrounded by walkers, he is thrown from his horse and ends up in a tank. This is where someone on the walkie talkie, Glenn, helps him out of the tight situation.',
      season: 1,
      episode: 2,
      episode_title: 'Guts',
      imdb: 'https://www.imdb.com/title/tt1628064/?ref_=ttep_ep2',
      url: 'https://www.youtube.com/watch?v=ZxmWhlidAZ8',
      thumbnail: './assets/images/walking-dead.jpg'
    },
    {
      description:
        'The Saviours have captured Rick and the group. They kneel at the mercy of Negan who makes an example of 2 of the groups members. Not for the faint of heart.',
      season: 7,
      episode: 1,
      episode_title: "The Day Will Come When You Won't Be",
      imdb: 'https://www.imdb.com/title/tt5164772/?ref_=ttep_ep1',
      url: 'https://www.youtube.com/watch?v=d3LDG5CCOPQ',
      thumbnail: './assets/images/walking-dead.jpg'
    }
  ]
};

it('renders without crashing, with zero items', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MultiCard {...itemMockZero} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing, with one item', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MultiCard {...itemMockOne} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing, with two items', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MultiCard {...itemMockTwo} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot of initial component with multiple items', () => {
  let render = createRender();
  const component = render(<MultiCard {...itemMockTwo} />);
  expect(component).toMatchSnapshot();
});
