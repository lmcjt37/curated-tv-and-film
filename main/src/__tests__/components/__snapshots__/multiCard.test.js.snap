// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of initial component with multiple items 1`] = `
<div
  className="item-card-multi"
>
  <div
    className="item-header"
  >
    <h2>
      The Walking Dead
      
    </h2>
    <span
      className="genre-container"
    >
      <h5
        className="genre-title"
      >
        Genre:
      </h5>
      <span
        className="genre-item"
      >
        Drama, Horror, Action
      </span>
    </span>
  </div>
  <div
    className="item-card__row "
  >
    <div
      className="item-thumbnail"
    >
      <a
        href="https://www.youtube.com/watch?v=ZxmWhlidAZ8"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          className="item-image"
          src="./assets/images/walking-dead.jpg"
        />
      </a>
    </div>
    <div
      className="item-info"
    >
      <div
        className="tvshow-details"
      >
        <div
          className="details__item"
        >
          <span>
            Season:
          </span>
           
          1
        </div>
        <div
          className="details__item"
        >
          <span>
            Episode:
          </span>
           
          2
        </div>
        <div
          className="details__item"
        >
          <span>
            Episode Title:
          </span>
           
          Guts
        </div>
      </div>
      <div
        className="item-desc details__item"
      >
        <span>
          Description:
        </span>
         
        Following on from where Rick Grimes wakes up from a coma and tries to find his family. He gets to Atlanta, surrounded by walkers, he is thrown from his horse and ends up in a tank. This is where someone on the walkie talkie, Glenn, helps him out of the tight situation.
      </div>
      <div
        className="item-imdb"
      >
        <span>
          IMDB:
        </span>
        <a
          className="item-link"
          href="https://www.imdb.com/title/tt1628064/?ref_=ttep_ep2"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://www.imdb.com/title/tt1628064/?ref_=ttep_ep2
        </a>
      </div>
    </div>
  </div>
  <div
    className="item-card__row item-card__hr"
  >
    <div
      className="item-thumbnail"
    >
      <a
        href="https://www.youtube.com/watch?v=d3LDG5CCOPQ"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          className="item-image"
          src="./assets/images/walking-dead.jpg"
        />
      </a>
    </div>
    <div
      className="item-info"
    >
      <div
        className="tvshow-details"
      >
        <div
          className="details__item"
        >
          <span>
            Season:
          </span>
           
          7
        </div>
        <div
          className="details__item"
        >
          <span>
            Episode:
          </span>
           
          1
        </div>
        <div
          className="details__item"
        >
          <span>
            Episode Title:
          </span>
           
          The Day Will Come When You Won't Be
        </div>
      </div>
      <div
        className="item-desc details__item"
      >
        <span>
          Description:
        </span>
         
        The Saviours have captured Rick and the group. They kneel at the mercy of Negan who makes an example of 2 of the groups members. Not for the faint of heart.
      </div>
      <div
        className="item-imdb"
      >
        <span>
          IMDB:
        </span>
        <a
          className="item-link"
          href="https://www.imdb.com/title/tt5164772/?ref_=ttep_ep1"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://www.imdb.com/title/tt5164772/?ref_=ttep_ep1
        </a>
      </div>
    </div>
  </div>
</div>
`;
